doc/overlays.xml

```diff
@@ -8,59 +8,62 @@
 overlays. Overlays are used to add layers in the fix-point used by Nixpkgs
 to compose the set of all packages.</para>
 
+<para>Nixpkgs can be configured with a list of overlays, which are
+applied in order. This means that the order of the overlays can be significant
+if multiple layers override the same package.</para>
+
 <!--============================================================-->
 
 <section xml:id="sec-overlays-install">
-<title>Installing Overlays</title>
+<title>Installing overlays</title>
 
-<para>The set of overlays is looked for in the following places. The
-first one present is considered, and all the rest are ignored:
+<para>The list of overlays is determined as follows:
 
 <orderedlist>
 
   <listitem>
+    <para>First, if an <varname>overlays</varname> argument to the nixpkgs function itself is given,
+    then that is used. This can be passed explicitly when importing nipxkgs, for example 
+    <literal>import &lt;nixpkgs> { overlays = [ overlay1 overlay2 ] }</literal>.</para>
 
-    <para>As an argument of the imported attribute set. When importing Nixpkgs,
-    the <varname>overlays</varname> attribute argument can be set to a list of
-    functions, which is described in <xref linkend="sec-overlays-layout"/>.</para>
-
+    <para>On a NixOS system the value of the <literal>nixpkgs.overlays</literal> option, if present, 
+    is passed to the system Nixpkgs in this way. Note that this does not affect the overlays for
+    non-NixOS operations (e.g. <literal>nix-env</literal>), which are looked up independently.</para>
   </listitem>
 
   <listitem>
+    <para>Otherwise, if the Nix path entry <literal>&lt;nixpkgs-overlays></literal> exists and is a 
+    directory, then the result is the set of overlays found in that directory, ordered lexicographically.</para> 
 
-    <para>In the directory pointed to by the Nix search path entry
-    <literal>&lt;nixpkgs-overlays></literal>.</para>
+    <para>See the section on <literal>NIX_PATH</literal> in the Nix manual for more details on how to 
+    set a value for <literal>&lt;nixpkgs-overlays>.</literal></para>
   </listitem>
 
   <listitem>
-
-    <para>In the directory <filename>~/.config/nixpkgs/overlays/</filename>.</para>
+    <para>Otherwise, if <filename>~/.config/nixpkgs/overlays/</filename> exists and is a directory, then
+    the result is the set of overlays found in that directory, ordered lexicographically.</para>
   </listitem>
 
 </orderedlist>
 </para>
 
-<para>For the second and third options, the directory should contain Nix expressions defining the
-overlays. Each overlay can be a file, a directory containing a
-<filename>default.nix</filename>, or a symlink to one of those. The expressions should follow
-the syntax described in <xref linkend="sec-overlays-layout"/>.</para>
+<para>For the second and third options overlays can be provided as files, 
+directories containing a <filename>default.nix</filename>, or symlinks to one of those.</para>
```

Comment:
nit: For the second and third options*,* overlays are registered in the given directory as files, directories containing …